answerking_app/views/category_views.py

```diff
@@ -16,33 +17,31 @@
 
 class CategoryListView(View):
     @csrf_exempt
-    def get(self, request: HttpRequest, *args, **kwargs) -> JsonResponse | HttpResponse:
+    def get(self, request: Request, *args, **kwargs) -> Response:
         categories: QuerySet[Category] = category_service.get_all()
         response: list[ReturnDict] = []
 
         if categories:
             response = [CategorySerializer(cat).data for cat in categories]
 
-        return JsonResponse(response, encoder=DjangoJSONEncoder, safe=False)
+        return Response(response, status=status.HTTP_200_OK)
 
     @csrf_exempt
-    def post(
-        self, request: HttpRequest, *args, **kwargs
-    ) -> JsonResponse | HttpResponse:
+    def post(self, request: Request, *args, **kwargs) -> Response:
         try:
             body: dict | None = json.loads(request.body)
```

Comment:
body type is set to dict here, but in the service it expects CategoryDict. There's several other pyright errors similar to this across item and order views too

Reply:
All errors have been sorted. It's also worth pointing out that pyright doesn't like it when you put an f-string as the key of a dict and throws a generic error. Just something to be aware of.